src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,120 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+const initialMovieState = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+interface NewMovieProps {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<NewMovieProps> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [isSubmitActive, setIsSubmitActive] = useState(false);
+  const [movie, setMovie] = useState<Movie>(initialMovieState);
+
+  const handleInputChange = (key: string, value: string) => {
+    setMovie((prevState) => {
+      const newMovie = {
+        ...prevState,
+        [key]: value,
+      };
+
+      const [title, , imgUrl, imdbUrl, imdbId]
+        = [...Object.values(newMovie).map(entrie => entrie.trim())];
+
+      if (title.length > 0
+        && imgUrl.length > 0
+        && imdbUrl.length > 0
+        && imdbId.length > 0
+      ) {
+        setIsSubmitActive(true);
+      } else {
+        setIsSubmitActive(false);
+      }
```

Comment:
```suggestion

        setIsSubmitActive(title.length > 0
          && imgUrl.length > 0
          && imdbUrl.length > 0
          && imdbId.length > 0);
```
looks like If is redundant here